pkg/cloud/services/networking/trunk.go

```diff
@@ -77,6 +79,44 @@ func (s *Service) getOrCreateTrunkForPort(eventObject runtime.Object, port *port
 	return trunk, nil
 }
 
+func (s *Service) RemoveTrunkSubports(trunkID string) error {
+	mc := metrics.NewMetricPrometheusContext("trunk", "deletesubports")
+	subports, err := s.client.ListTrunkSubports(trunkID)
+
+	if mc.ObserveRequestIgnoreNotFound(err) != nil {
+		return err
+	}
+
+	if len(subports) < 1 {
+		return nil
+	}
+
+	portList := make([]trunks.RemoveSubport, len(subports))
+	for i, subport := range subports {
+		portList[i] = trunks.RemoveSubport{
+			PortID: subport.PortID,
+		}
+	}
+
+	removeSubportsOpts := trunks.RemoveSubportsOpts{
+		Subports: portList,
+	}
+
+	err = s.client.RemoveSubports(trunkID, removeSubportsOpts)
+	if mc.ObserveRequest(err) != nil {
+		return err
+	}
+
+	for _, subPort := range subports {
+		err := s.client.DeletePort(subPort.PortID)
```

Comment:
Just a thought: does deleting the subport also remove it from the trunk? It normally works that way in other contexts, but I haven't checked. If so we could potentially skip the RemoveSubports call above.

Reply:
I haven't thought about it, but that sounds reasonable

Reply:
Hi @mdbooth, I have asked around and gave this some thought, and imo it is a good practice to keep the explicit deletion of subports from the trunk. It's possible that even after the ports are deleted, the trunk still has reference to them, hence it'd mess up, at least from the log perspectives. Also, it's more graceful that we first disconnect the trunk from the subports, then remove the sub ports, than directly removing them. Wdyt?

Reply:
WFM.